txnbuild/helpers.go

```diff
@@ -192,3 +192,20 @@ func NewValidationError(field, message string) *ValidationError {
 		Message: message,
 	}
 }
+
+// Parses an asset string in canonical form (SEP-11) into an Asset structure.
+// https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset
+func ParseAssetString(canonical string) (Asset, error) {
+	assets, err := xdr.BuildAssets(canonical)
+	if err != nil {
+		return &NativeAsset{}, errors.Wrap(err, "failed parsing: is it in canonical (SEP-11) form?")
+	}
+
+	// It returns a list, so we'll need to grab the first element.
+	asset, err := assetFromXDR(assets[0])
+	if err != nil {
+		return &NativeAsset{}, errors.Wrap(err, "failed to create Asset from XDR")
```

Comment:
This error centers on the issue of XDR, but the caller to this function probably won't be expecting XDR to be involved. That's an implementation detail. We can mention that, but should call out this is happening as part of trying to parse the asset.
```suggestion
		return &NativeAsset{}, errors.Wrap(err, "error parsing asset string via XDR types")
```